test/gtest/common/test.h

```diff
@@ -0,0 +1,250 @@
+/**
+ * Copyright (C) Mellanox Technologies Ltd. 2001-2014.  ALL RIGHTS RESERVED.
+ * Copyright (C) Huawei Technologies Co., Ltd. 2020.  All rights reserved.
+ *
+ * See file LICENSE for terms.
+ */
+
+#ifndef UCC_TEST_BASE_H
+#define UCC_TEST_BASE_H
+
+/* gcc 4.3.4 compilation */
+#ifndef UINT8_MAX
+#define __STDC_LIMIT_MACROS
+#include <stdint.h>
+#endif
+
+#define __STDC_FORMAT_MACROS 1
+#include <inttypes.h>
+
+#include "test_helpers.h"
+
+#include <core/ucc_global_opts.h>
+
+#include <map>
+#include <vector>
+#include <string>
+
+namespace ucc {
+
+/**
+ * Base class for tests
+ */
+class test_base {
+public:
+    typedef enum {
+        IGNORE_IF_NOT_EXIST,
+        FAIL_IF_NOT_EXIST,
+        SETENV_IF_NOT_EXIST,
+        SKIP_IF_NOT_EXIST
+    } modify_config_mode_t;
+
+    test_base();
+    virtual ~test_base();
+
+protected:
+    typedef enum {
+        NEW, RUNNING, SKIPPED, ABORTED, FINISHED
+    } state_t;
+
+    typedef std::vector<ucc_global_config_t> config_stack_t;
+
+    void SetUpProxy();
```

Comment:
what is the user acse for Proxy ? How is it used?

Reply:
In general the Proxy is there to collect errors and warnings instead of giving it to gtest - which would likely fail on an error and stop everything...